Cinema system/Cinema system/Services/CinemaService.cs

```diff
@@ -106,35 +105,31 @@ public int CreateHall(int cinemaId, HallView hall)
                         TypeOfSeat = (SeatType)Enum.Parse(typeof(SeatType), s.TypeOfSeat),
                         Row = s.Row,
                         Place = s.Place
-                    }).ToList()
+                    })
+                    .ToList()
             };
 
-            var cinema = _db.Cinemas
-                 .Include(c => c.Halls)
-                 .FirstOrDefault(c => c.Id == cinemaId);
-
-            cinema.Halls.Add(newHall);
+            _db.Halls.Add(newHall);
 
             _db.SaveChanges();
 
             return newHall.Id;
         }
 
-        public void EditHall(int hallId, HallView hall)
+        public void EditHall(int hallId, HallFullView hall)
         {
-            var editHall = _db.Halls
-                .Include(h => h.Seats)
-                .FirstOrDefault(h => h.Id == hallId);
+            var editHall = _db.Halls.Find(hallId);
```

Comment:
What about Include()? Is it required? Have you considered using SingleOrDefault()?